ceremonies/src/lib.rs

```diff
@@ -515,101 +748,95 @@ impl<T: Config> Module<T> {
-                    }
-
-                    if was_attested_count < (n_honest_participants - 1)
-                    {
-                        debug!(
-                            "skipped participant because of too few attestations ({}): {:?}",
-                            was_attested_count,
-                            p
-                        );
-                        continue;
-                    }
-
-                    trace!(target: LOG, "participant merits reward: {:?}", p);
-                    if <encointer_balances::Module<T>>::issue(*cid, &p, reward).is_ok() {
-                        <ParticipantReputation<T>>::insert(
-                            (cid, cindex),
-                            &p,
-                            Reputation::VerifiedUnlinked,
-                        );
-                    }
+
+        let cindex = <encointer_scheduler::Module<T>>::current_ceremony_index() - 1;
+        let reward = Self::nominal_income(cid);
+        let m = Self::get_meetup_index((*cid, cindex), participant)?;
+
+        if <IssuedRewards>::contains_key((cid, cindex), m) {
+            return Err(<Error<T>>::RewardsAlreadyIssued.into());
+        }
+
+        // first, evaluate votes on how many participants showed up
+        let (n_confirmed, n_honest_participants) =
+            match Self::ballot_meetup_n_votes(cid, cindex, m) {
+                Some(nn) => nn,
+                _ => {
+                    return Err(<Error<T>>::VotesNotDependable.into());
+                }
+            };
+        let meetup_participants = Self::get_meetup_participants((*cid, cindex), m);
+        for p in &meetup_participants {
+            if Self::meetup_participant_count_vote((cid, cindex), &p) != n_confirmed {
+                debug!(
+                    target: LOG,
+                    "skipped participant because of wrong participant count vote: {:?}",
+                    p
+                );
+                continue;
+            }
+            let attestees = Self::attestation_registry(
+                (cid, cindex),
+                &Self::attestation_index((*cid, cindex), &p),
+            );
+            if attestees.len() < (n_honest_participants - 1) as usize {
+                debug!(
+                    target: LOG,
+                    "skipped participant because didn't testify for honest peers: {:?}",
+                    p
+                );
+                continue;
+            }
+
+            let mut was_attested_count = 0u32;
+            for other_participant in &meetup_participants {
+                if other_participant == p {
+                    continue;
+                }
+                let attestees_from_other = Self::attestation_registry(
+                    (cid, cindex),
+                    &Self::attestation_index((cid, cindex), &other_participant),
+                );
+                if attestees_from_other.contains(&p) {
+                    was_attested_count += 1;
                 }
```

Comment:
nested loops have high potential of confusion.. the operation you're doing here is a good use case for the iter() api..
pseudo-code: 
let was_attested_count = meetup_participants.iter()
.filter(o -> p!=o)
.map(o -> Self::attestation_index(...))
.map(o -> Self::attestation_registry(...))
.filter(o -> attestees_from_other(p))
.count();